src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -801,6 +809,15 @@ void GcsActorManager::DestroyActor(const ActorID &actor_id,
         // Destroy placement group owned by this actor.
         destroy_owned_placement_group_if_needed_(actor_id);
       }));
+
+  // Inform all creation callbacks that the actor was cancelled, not created.
+  auto iter = actor_to_create_callbacks_.find(actor_id);
```

Comment:
Since the same logic exists when `OnActorSchedulingSuccess`, I think it's better to move this logic from `DestroyActor` to `OnActorSchedulingFailed`.